File: cpp/f-1.cpp
#include <algorithm> // min, max, swap, sort, reverse, lower_bound, upper_bound
#include <bitset>    // bitset
#include <cctype>    // isupper, islower, isdigit, toupper, tolower
#include <cstdint>   // int64_t, int*_t
#include <cstdio>    // printf
#include <deque>     // deque
#include <iostream>  // cout, endl, cin
#include <map>       // map
#include <queue>     // queue, priority_queue
#include <set>       // set
#include <stack>     // stack
#include <string>    // string, to_string, stoi
#include <tuple>     // tuple, make_tuple
#include <unordered_map> // unordered_map
#include <unordered_set> // unordered_set
#include <utility>       // pair, make_pair
#include <vector>        // vector

using namespace std;
typedef long long ll;

using P = pair<int, int>;

int main() { 
    string s;
    cin >> s;

    map<char, int> mp;

    for (int i = 0; i < s.size(); i++) {
        mp[s[i]]++;
    }

    map<int, int> mp2;

    for (auto p : mp) {
       mp2[mp[p.first]]++;
    }
    
    for (auto p : mp2) {
       if (p.second != 2) {
            cout << "No" << endl;
            return 0;
       }
    }

    cout << "Yes" << endl;

    return 0; 
}

File: python/f-1.py
def main():
    s = input()

    mp = {}
    for c in s:
        if c in mp:
            mp[c] += 1
        else:
            mp[c] = 1

    mp2 = {}
    for cnt in mp.values():
        if cnt in mp2:
            mp2[cnt] += 1
        else:
            mp2[cnt] = 1

    for cnt in mp2.values():
        if cnt != 2:
            print("No")
            return
    
    print("Yes")

if __name__ == "__main__":
    main()
